### DISCUSSION TITLE
Creating Datatables from pygetpapers

### DISCUSSION THREAD

**Discussion by rkclimate20** [semanticClimate/ipcc-report-cataloguing#16] [General]
A semantic corpus of documents either:
* from searching the literature with pygetpapers 
* or from a government or an International organization etc. 

convert corpus into semantic form. major component is a  jQuery datatables. We have a list of the identifiers of the documents. 
- This has identifier for the paper from the publisher by the DOI.
-  Some metadata for authors, journals and abstract.

**Comment by petermr** [comment]
# Creating `datatables` from `pygetpapers`

* Ideally should be part of `pygetpapers`
* current workaround uses `amilib`
prototype in
`amilib.test_bib::PygetpapersTest::test_make_html_table_from_json()`

```
 def test_make_html_table_from_json(self):
        """
        assume JSON is an implicit table
        """

        project_name = "district_heating"
        wanted_keys = [PMCID, DOI, TITLE, AUTHOR_STRING, JOURNAL_INFO_TITLE, ABS_TEXT]

        project_dir = Path(Resources.TEST_RESOURCES_DIR, "json", project_name)
        config_ini = Path(project_dir, SAVED_CONFIG_INI)
        infile = Path(project_dir, EUPMC_RESULTS_JSON)
        outdir = Path(Resources.TEMP_DIR, "json", project_name)

        outdir.mkdir(parents=True, exist_ok=True)
        outfile_h = Path(outdir, "europe_pmc.html")
        datatables = True

        Pygetpapers.read_json_create_write_html_table(
            infile, outfile_h, wanted_keys, datatables=datatables, table_id=None, config_ini=config_ini)
```
